pkg/datapath/linux/probes/managed_neighbors.go

```diff
@@ -0,0 +1,84 @@
+package probes
+
+import (
+	"fmt"
+	"net"
+	"runtime"
+
+	"github.com/vishvananda/netlink"
+	"github.com/vishvananda/netns"
+)
+
+// HaveManagedNeighbors creates a netlink.Neigh entry with the flag NTF_EXT_MANAGED.
+func HaveManagedNeighbors() error {
+	ch := make(chan bool)
+	var err error
+
+	// In order to call HaveManagedNeighbors safely, it has to be started
+	// in a goroutine, so we can make sure the goroutine ends when the function exits.
+	// This gives us control over the lifecycle of the underlying thread.
+	go func() {
+		err = haveManagedNeighbors()
+		ch <- true
+	}()
+	<-ch // wait for probe to finish
+
+	return err
+}
+
+func haveManagedNeighbors() (outer error) {
+	runtime.LockOSThread()
```

Comment:
Hmm, that's by design. If we defer the unlock here, we can't stop the OS thread from becoming schedulable again if netns.Set fails. Using Goexit instead of return wouldn't help either, since it runs all deferred functions before terminating the goroutine.

Reply:
Thanks for the explanation. I just noticed https://github.com/cilium/cilium/pull/25134#discussion_r1189953611 now which explains the rationale behind this. I think this should at least be briefly mentioned in a comment so others reading the code later on don't stumble upon this.

Reply:
Another option is calling `runtime.LockOSThread(); defer runtime.UnlockOSThread()` here and adding an extra `LockOSThread()` after checking the error from `netns.New()`.

In order for an OS thread to be unlocked, as many calls to `Unlock()` need to be made as there were calls to `Lock()`.

Reply:
I left it the way it was, but added a comment to explain the late `defer runtime.UnlockOSThread()`.